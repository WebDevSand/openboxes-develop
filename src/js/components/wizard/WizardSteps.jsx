import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import './WizardSteps.scss';

const WizardSteps = props => (
  <div className="steps-box d-print-none">
    <div className="steps d-flex flex-wrap">
      { _.map(props.steps, (step, index) => (
        <div
          key={index}
          className="step-container"
        >
          <button className="step" onClick={() => props.onClick(index + 1)} disabled={!props.stepsClickable}>
            <i
              className={`fa 
                ${index + 1 === props.currentStep ? 'fa-circle-o' : ''} 
                ${index + 1 < props.currentStep ? 'fa-check-circle-o' : ''} 
                ${index + 1 > props.currentStep || (index + 1 !== props.currentStep && !props.stepsClickable) ? 'fa-circle-o disabled' : ''}`}
              aria-hidden="true"
            />
            <span className="step-text">
              {step}
            </span>
          </button>
        </div>
          ))
        }
    </div>
  </div>
);

export default WizardSteps;

WizardSteps.propTypes = {
  /** Array of steps names */
  steps: PropTypes.arrayOf(PropTypes.string).isRequired,
  /** Current step number */
  currentStep: PropTypes.number.isRequired,
  /** Function called after clicking on step, (non required) */
  /* eslint-disable-next-line react/no-unused-prop-types */
  onClick: PropTypes.func,
  /** Indicator if steps are clickable (default = false) */
  stepsClickable: PropTypes.bool,
};

WizardSteps.defaultProps = {
  onClick: stepIdx => stepIdx,
  stepsClickable: false,
};
