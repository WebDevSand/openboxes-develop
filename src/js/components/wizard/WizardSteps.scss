.steps-box {
    padding: 12px 24px 24px;
    display: flex;
    background-color: #fff;
    margin: 0.5rem 1.5rem 0;
    box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.09);

    .steps {
        margin: auto;
        flex: 1;
        justify-content: space-evenly;

        .step-container {
            display: flex;

            &:not(:last-child) {
                flex: 1;
            }

            .step {
                border: none;
                background: none;
                display: flex;
                flex-direction: column;
                align-items: center;
                position: relative;

                // Icons
                .fa {
                    font-size: 20px !important;
                }

                .fa-check-circle-o {
                    color: #50c146;
                }

                .fa-circle-o.disabled {
                    color: #e9ebeb;
                }

                .fa-circle-o {
                    color: #4f87ce;
                }

                .step-text {
                    position: absolute;
                    top: 20px;
                    font-size: small;
                    white-space: nowrap;
                }
            }
        }

        .step-container:not(:last-child)::after {
            content: " ";
            border-bottom: 1px dashed #e9ebeb;
            flex: 1;
            height: 18px;
        }

    }
}
