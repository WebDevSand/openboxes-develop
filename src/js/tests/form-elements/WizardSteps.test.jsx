import React from 'react';
import renderer from 'react-test-renderer';
import WizardSteps from '../../components/wizard/WizardSteps';

it('WizardSteps renders correctly', () => {
  const steps = ['Create', 'Add items', 'Edit', 'Pick', 'Send'];
  const currentStep = 2;
  const component = renderer
    .create(<WizardSteps steps={steps} currentStep={currentStep} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
