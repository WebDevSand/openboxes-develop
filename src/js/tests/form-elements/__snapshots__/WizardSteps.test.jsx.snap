// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardSteps renders correctly 1`] = `
<div
  className="steps-box d-print-none"
>
  <div
    className="steps d-flex flex-wrap"
  >
    <div
      className="step-container"
    >
      <button
        className="step"
        disabled={true}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa 
                 
                fa-check-circle-o 
                fa-circle-o disabled"
        />
        <span
          className="step-text"
        >
          Create
        </span>
      </button>
    </div>
    <div
      className="step-container"
    >
      <button
        className="step"
        disabled={true}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa 
                fa-circle-o 
                 
                "
        />
        <span
          className="step-text"
        >
          Add items
        </span>
      </button>
    </div>
    <div
      className="step-container"
    >
      <button
        className="step"
        disabled={true}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa 
                 
                 
                fa-circle-o disabled"
        />
        <span
          className="step-text"
        >
          Edit
        </span>
      </button>
    </div>
    <div
      className="step-container"
    >
      <button
        className="step"
        disabled={true}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa 
                 
                 
                fa-circle-o disabled"
        />
        <span
          className="step-text"
        >
          Pick
        </span>
      </button>
    </div>
    <div
      className="step-container"
    >
      <button
        className="step"
        disabled={true}
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa 
                 
                 
                fa-circle-o disabled"
        />
        <span
          className="step-text"
        >
          Send
        </span>
      </button>
    </div>
  </div>
</div>
`;
